static/js/stream_ui_updates.js

```diff
@@ -114,9 +114,10 @@ exports.update_stream_row_in_settings_tab = function (sub) {
     // stream row under tab.
     if (subs.is_subscribed_stream_tab_active()) {
         const sub_row = subs.row_for_stream_id(sub.stream_id);
+        const tab_key = subs.get_active_data().tab.attr("data-tab-key");
         if (sub.subscribed) {
             sub_row.removeClass("notdisplayed");
-        } else if (sub.invite_only || page_params.is_guest) {
+        } else if ((sub.invite_only || page_params.is_guest) && tab_key !== 'all-streams') {
```

Comment:
Would it make more sense for the `tab-key` conditional to be an early-return check before this if/else block?  

Oh, it is: Isn't `subs.is_subscribed_stream_tab_active` above supposed to check for this?

Reply:
Oh yeah, I hadn't pay attention to that sorry, I focused only on why it was entering the else if. But I got it. It's wrong because the selector finds two elements, the new sort button, and the tab switch.

![oie_0ViilHgLP1vj](https://user-images.githubusercontent.com/15564640/81631645-afd1b380-93de-11ea-8244-d815c3cb1c13.png)

That sort icon is selected so, the function will return true. I'm fixing it.